ipfs/gateway/nginx.conf

```diff
@@ -167,7 +167,7 @@ server {
         return 404;
     }
 
-    rewrite "^(.*)$" \$ipfspath\$1;
+    rewrite "^(.*)\$" \$ipfspath\$1;
```

Comment:
Just as a sanity check, are you *sure* that escape is needed? Is that because we're quoting? I'm definitely not an NGINX expert but the docs don't say anything about escaping (actually, in this case, they recommend using a `return`):

https://www.nginx.com/blog/creating-nginx-rewrite-rules/

Reply:
Yes -- that thing is piped through our two-liner bash template engine. It evaluates templates as bash code, and thus variable signs need to be escaped. (I know, very prone to error, and one more reason to get rid of the provsn tool.)

Reply:
Oh. m4? Oh well...

Reply:
No, much more stupid :) https://github.com/ipfs/infrastructure/blob/e376e681774e2a1b7627d44e2b05184c54f0476c/provsn#L6-L28 -- could actually do that escaping the template function though, similar to how double-quotes are being escaped automatically.

Reply:
Ow!